app/src/main/java/com/example/coopertest/Results.kt

```diff
@@ -0,0 +1,163 @@
+package com.example.coopertest
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.provider.Settings.Global.getString
+import androidx.preference.PreferenceManager
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.*
+
+class Results (meters : Double, context: Context){
+
+    var meters : Double=0.0
+    var age: Int=0
+    var athlete: Boolean = false
+    var gender : String = ""
+
+    init {
+        val mSharedPreference = PreferenceManager.getDefaultSharedPreferences(context)
+        val dateBirthday = mSharedPreference.getString("birthday", "null")
+        age=getAge(dateBirthday.toString())
+        this.meters=meters
+        athlete = mSharedPreference.getBoolean("athlete", false)
+        gender = mSharedPreference.getString("gender", "Male")!!
+    }
+
+    private fun getAge(dobString: String): Int {
+
+        var date: Date? = null
+        val sdf = SimpleDateFormat("yyyy-MM-dd")
+        try {
+            date = sdf.parse(dobString)
+        } catch (e: ParseException) {
+            e.printStackTrace()
+        }
+
+        if (date == null) return 0
+
+        val dob = Calendar.getInstance()
+        val today = Calendar.getInstance()
+
+        dob.setTime(date)
+
+        val year = dob.get(Calendar.YEAR)
+        val month = dob.get(Calendar.MONTH)
+        val day = dob.get(Calendar.DAY_OF_MONTH)
+
+        dob.set(year, month + 1, day)
+
+        var age = today.get(Calendar.YEAR) - dob.get(Calendar.YEAR)
+
+        if (today.get(Calendar.DAY_OF_YEAR) < dob.get(Calendar.DAY_OF_YEAR)) {
+            age--
+        }
+
+        return age
+    }
+
+    fun getLevel(): String {
+        var yourResult : String=""
+        if (athlete==true) {
```

Comment:
This shouldn't be done like that. Better solution - use when. Even better - use dictionary/map. But probably it should be a file in the resources, with all the values needed. And also the strings should be in strings.xml. Now it's all hard-coded and very very long

Reply:
I changed the string and I put in string values after I though to use another system, but it was complicated: it's like a table with 3 variables

Reply:
I forgot to mention by the way, I think there's a mathematical formula that calculates your score, it might simplify things

Reply:
And for the strings you need to write getString(R.string.good)

Reply:
Wow ! I should be very good !

Reply:
I think you could use a map here, that maps Tuple (gender, age, athlete) to a list of thresholds for the certain result (0, 1 etc. and later map that to very bad, bad etc... )
This way you remove all the ifs, can see all the thresholds in one place and don't need to traverse all this if tree, and actually have notion of order, e.g. that bad is between very bad and average

Reply:
And if we don't write the two neighbouring levels, we should write the bounds. Like "Good (2300-2700m)"